Sources/Common/Background Queue/QueueStorage.swift

```diff
@@ -61,17 +68,22 @@ public class FileManagerQueueStorage: QueueStorage {
         self.logger = logger
         self.dateUtil = dateUtil
         self.lock = lockManager.getLock(id: .queueStorage)
+        self.inventoryStore = inventoryStore
     }
 
     public func getInventory() -> [QueueTaskMetadata] {
         lock.lock()
         defer { lock.unlock() }
 
-        guard let data = fileStorage.get(type: .queueInventory, fileId: nil) else { return [] }
+        if let inventory = inventory {
+            return inventory
+        }
 
-        let inventory: [QueueTaskMetadata] = jsonAdapter.fromJson(data) ?? []
+        guard let data = fileStorage.get(type: .queueInventory, fileId: nil) else { return [] }
+        let readInventory: [QueueTaskMetadata] = jsonAdapter.fromJson(data) ?? []
+        inventory = readInventory
 
-        return inventory
+        return readInventory
```

Comment:
The class is designed so that if you want the inventory, you need to call `getInventory()` because of the `lock` inside of `getInventory()`. However, I can see how someone could accidentally access the `inventory` property of the class. Adding a comment to `inventory` saying to use `getInventory()` may be able to help here. 

Would this satisfy your concern?

Reply:
Yes, it does and it looks much cleaner and fail-proof, thank you.